wild-swimming-app/src/pages/location-details/Location-details.tsx

```diff
@@ -1,5 +1,61 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import useBathingWaterRequest from "../../hooks/useBathingWaterRequest";
+import { LocationDetailsT } from "../../types/types";
+
+import styles from "./location-details.module.css";
+
+type LocationResponseT<T> = {
+  location: Array<T>;
+};
+
 const LocationDetails = () => {
-  return <section></section>;
+  const [location, setLocation] = useState<LocationDetailsT>();
+
+  const { id } = useParams();
+
+  const { sendLocationRequest } = useBathingWaterRequest();
+
+  useEffect(() => {
+    const res = (data: LocationResponseT<LocationDetailsT>) => {
+      setLocation(data.location[0]);
+    };
+
+    sendLocationRequest(
+      "GET",
+      `https://environment.data.gov.uk/data/bathing-water-profile/${id}/2023:1`,
+      res
+    );
+  }, [sendLocationRequest, id]);
+
+  console.log(location);
```

Comment:
keeping for later? just making comment so we remember to take it out.

Reply:
aye, just wanted to push this so people can use the new router code quicker, will remove once I've done page this aft. Thanks, Nico!